src/communication/stmt.rs

```diff
@@ -0,0 +1,153 @@
+use super::CommunicationReplacer;
+use crate::comm_repl_default_impl;
+use crate::{ast::*, ClassOrInterfaceComponent};
+use crate::{prophet::ModuleComponent, MethodComponent};
+
+comm_repl_default_impl!(
+    ReturnStmt,
+    ImportStmt,
+    BreakStmt,
+    ContinueStmt,
+    ThrowStmt,
+    CatchStmt
+);
+
+impl CommunicationReplacer for DeclStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        for expr in self.expressions.iter_mut() {
+            if let Some(Node::Expr(replacement)) =
+                expr.replace_communication_call(modules, module, class, method)
+            {
+                *expr = replacement;
+            }
+        }
+        None
+    }
+}
+
+impl CommunicationReplacer for ExprStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.expr
+            .replace_communication_call(modules, module, class, method)
+    }
+}
+
+impl CommunicationReplacer for IfStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
+            .replace_communication_call(modules, module, class, method);
+        if let Some(else_body) = self.else_body.as_mut() {
+            else_body.replace_communication_call(modules, module, class, method);
+        }
+        None
+    }
+}
+
+impl CommunicationReplacer for ForStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
+            .replace_communication_call(modules, module, class, method)
+    }
+}
+impl CommunicationReplacer for ForRangeStmt {
+    fn replace_communication_call(
+        &mut self,
+        modules: &Vec<ModuleComponent>,
+        module: &ModuleComponent,
+        class: Option<&ClassOrInterfaceComponent>,
+        method: &MethodComponent,
+    ) -> Option<Node> {
+        self.body
```

Comment:
Same here as with standard for loop. Probably not a big enough chance to worry about, but not a chance I want us to take without thinking about it.